fdbserver/MoveKeys.actor.cpp

```diff
@@ -55,6 +55,12 @@ struct Shard {
 };
 
 bool shouldCreateCheckpoint(const UID& dataMoveId) {
+	// Physical shard move and checkpoints are known to cause non-determinism in sharded rocks
+	// storage engine.
+	// TODO: Make checkpoint deterministic
+	if (g_network->isSimulated() && !noUnseed) {
```

Comment:
Only set noUnseed to true if shouldCreateCheckpoint returns true (i.e. physical shard move).